server/play-server/src/main/scala/tapir/server/play/PlayRequestToRawBody.scala

```diff
@@ -0,0 +1,60 @@
+package tapir.server.play
+
+import java.io.{ByteArrayInputStream, File}
+import java.nio.charset.Charset
+
+import akka.stream.Materializer
+import akka.util.ByteString
+import play.api.mvc.{PlayBodyParsers, RawBuffer, Request}
+import play.core.parsers.Multipart
+import tapir.model.Part
+import tapir.{
+  ByteArrayValueType,
+  ByteBufferValueType,
+  FileValueType,
+  InputStreamValueType,
+  MultipartValueType,
+  RawPart,
+  RawValueType,
+  StringValueType
+}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+class PlayRequestToRawBody(serverOptions: PlayServerOptions) {
+  def apply[R](rawBodyType: RawValueType[R], charset: Option[Charset], request: Request[RawBuffer], body: ByteString)(
+      implicit mat: Materializer
+  ): Future[R] = {
+    rawBodyType match {
+      case StringValueType(defaultCharset) => Future(new String(body.toArray, charset.getOrElse(defaultCharset)))
+      case ByteArrayValueType              => Future(body.toArray)
+      case ByteBufferValueType             => Future(body.toByteBuffer)
+      case InputStreamValueType            => Future(body.toArray).map(new ByteArrayInputStream(_))
+      case FileValueType =>
+        Future(java.nio.file.Files.write(serverOptions.temporaryFileCreator.create().path, body.toArray))
+          .map(p => p.toFile)
+      case mvt: MultipartValueType => multiPartRequestToRawBody(request, mvt)
+    }
+  }
+
+  private def multiPartRequestToRawBody[R](request: Request[RawBuffer], mvt: MultipartValueType)(
+      implicit mat: Materializer
+  ): Future[Seq[RawPart]] = {
+    val bodyParser = serverOptions.playBodyParsers.multipartFormData(
+      Multipart.handleFilePartAsTemporaryFile(serverOptions.temporaryFileCreator)
+    )
+    bodyParser.apply(request).run().flatMap {
+      case Left(value) => Future.failed(new RuntimeException(s"TODO handle it $value")) // what to do here?
```

Comment:
Why this happens while handling `multipart/form-data`? Should we use some other `BodyParser`?

Reply:
@zZKato any idea how to fix this? From my investigation it seems that play does some magic behind the scenes, before throwing it into multipartFormData body parser.

Reply:
@ghostbuster91 I think the problem here is that you handle the whole request as a FilePart even though we also have to consider DataParts (afaik).

Reply:
I'll investigate a bit further